File: microservices/purchase/src/main/kotlin/feri/si/vignette_purchase/logging/Log.kt
package feri.si.vignette_purchase.logging

import java.time.LocalDateTime
import java.util.*

data class Log(
    val logType: LogType,
    val http_call: String
){
    private val correlation_id: UUID = UUID.randomUUID();
    private val timestamp: String = LocalDateTime.now().toString();
    private val url: String  = "https://localhost:8080";
    private val app_name: String = "VignettePurchase";
    override fun toString(): String = "$timestamp $logType $url CorrelationId: $correlation_id [$app_name] - <* $http_call *>";
}

File: microservices/authentication/VignetteAuth/Logging/Log.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace VignetteAuth.Logging
{
    public class Log
    {
        public Guid CorrelationId { get; } = Guid.NewGuid();
        public DateTimeOffset Timestamp { get; } = DateTime.UtcNow;
        public LogType LogType { get; set; }
        public string Url { get; } = "https://localhost:44372";
        public string AppName { get; } = "VignetteAuth";
        public string HttpCall { get; set; }

        public Log(LogType logType, string httpCall)
        {
            LogType = logType;
            HttpCall = httpCall;
        }

        public override string ToString()
        {
            return string.Format("{0} {1} {2} Correlation: {3} [{4}] - <* {5} *>", Timestamp, LogType, Url, CorrelationId, AppName, HttpCall);
        }
    }
}
